nemo_curator/stages/deduplication/fuzzy/workflow.py

```diff
@@ -302,34 +311,43 @@ def run(
             minhash_end_time = time.time()
-            logger.info(f"Minhash pipeline completed in {(minhash_end_time - start_time):.2f} seconds")
+            minhash_time = minhash_end_time - minhash_start_time
+            workflow_result.add_pipeline_tasks("minhash", minhash_tasks)
+            workflow_result.add_metadata("minhash_time", minhash_time)
+            logger.info(f"Minhash pipeline completed in {minhash_time:.2f} seconds")
 
             lsh_pipeline = self._create_lsh_pipeline()
             lsh_start_time = time.time()
             # LSH stage generates it's own input tasks from the minhash directory
             lsh_tasks = lsh_pipeline.run(executor=executor, initial_tasks=None)
             lsh_end_time = time.time()
-            logger.info(f"LSH pipeline completed in {(lsh_end_time - lsh_start_time):.2f} seconds")
+            lsh_time = lsh_end_time - lsh_start_time
+            workflow_result.add_pipeline_tasks("lsh", lsh_tasks)
+            workflow_result.add_metadata("lsh_time", lsh_time)
+            logger.info(f"LSH pipeline completed in {lsh_time:.2f} seconds")
 
-            valid_lsh_tasks = [task for task in lsh_tasks if task._metadata.get("num_docs", 0) > 0]
+            valid_lsh_tasks = [task for task in lsh_tasks or [] if task._metadata.get("num_docs", 0) > 0]
             if len(valid_lsh_tasks) == 0:
                 logger.info("No potential duplicates found in the dataset. Skipping connected components pipeline.")
+                workflow_result.add_metadata("num_duplicates", 0)
             else:
                 connected_components_pipeline = self._create_connected_components_pipeline()
                 connected_components_start_time = time.time()
                 connected_components_tasks = connected_components_pipeline.run(
                     executor=executor, initial_tasks=valid_lsh_tasks
                 )
                 connected_components_end_time = time.time()
-                logger.info(
-                    f"Connected components pipeline completed in {(connected_components_end_time - connected_components_start_time):.2f} seconds"
-                )
+                connected_components_time = connected_components_end_time - connected_components_start_time
+                workflow_result.add_pipeline_tasks("connected_components", connected_components_tasks)
+                workflow_result.add_metadata("connected_components_time", connected_components_time)
+                logger.info(f"Connected components pipeline completed in {connected_components_time:.2f} seconds")
                 num_removed_documents = sum(
-                    task._metadata.get("num_removal_ids", 0) for task in connected_components_tasks
+                    task._metadata.get("num_removal_ids", 0) for task in (connected_components_tasks or [])
                 )
+                workflow_result.add_metadata("num_duplicates", num_removed_documents)
```

Comment:
Nit: call it `num_duplicates` to match the metadata name and the field name used in the exact deduplication workflow.
```suggestion
                workflow_result.add_metadata("num_duplicates", num_duplicates )
```